include/rocksdb/merge_operator.h

```diff
@@ -27,19 +27,28 @@ class Logger;
 //
 // To use merge, the client needs to provide an object implementing one of
 // the following interfaces:
-//  a) AssociativeMergeOperator - for most simple semantics (always take
-//    two values, and merge them into one value, which is then put back
-//    into rocksdb); numeric addition and string concatenation are examples;
+//  a) IdentityAssocMergeOperator - simplest interface for implementing
+//    merge semantics, always taking two inputs (values or merge operands)
+//    and merging them to one value.
+//    Requires: operator is associative, and no distinction between values
```

Comment:
May I suggest:

```
//    merge semantics; combines an input (value or merge operand) with a merge operand
//    to create a single result.
```

1. The `right` side arg is always a merge operand.
2. The output could be a value or merge_operand - there is no difference ;-)